pages/latest-books.vue

```diff
@@ -0,0 +1,178 @@
+          icon="i-heroicons-exclamation-circle"
+          color="orange"
+          variant="soft"
+          title="Join Our Affiliation Program!"
+          description="Setup your Liker ID and Stripe Connect Account to participate in sales affiliation program on Liker Land Bookstore."
+        />
+        <UButton @click="OnClickAffiliationSetup">
+          Go to Affiliation Setup
+        </UButton>
+      </template>
+      <template v-else>
+        Append <UKbd class="font-mono">
+          ?from={{ channelId }}
+        </UKbd> in any bookstore page to earn commission from book sales.
+      </template>
+    </UCard>
+    <UCard
+      :ui="{
+        header: { base: 'flex justify-between items-center' },
+        body: { padding: '' },
+        footer: { base: 'text-center' },
+      }"
+    >
+      <template #header>
+        <h2 class="text-xl font-bold font-mono">
+          Latest Books on LikerLand Bookstore
+        </h2>
+      </template>
+      <UTable
+        :columns="tableColumns"
+        :rows="tableRows"
+        @select="selectTableRow"
+      >
+        <template #image-data="{ row }">
+          <img
+            :src="row.image"
+            :alt="row.className"
+            class="w-12 h-12 object-cover rounded-lg"
+          >
+        </template>
+        <template #url-data="{ row }">
+          <UButton
+            label="Copy"
+            @click="e => copyToClipboard(e, row.url)"
+          />
+        </template>
+      </UTable>
+    </UCard>
+  </PageBody>
+</template>
+
+<script setup lang="ts">
+import { storeToRefs } from 'pinia'
+import { useBookStoreApiStore } from '~/stores/book-store-api'
+import { useUserStore } from '~/stores/user'
+import { useStripeStore } from '~/stores/stripe'
+import { useWalletStore } from '~/stores/wallet'
+
+const { LIKER_LAND_URL, LIKE_CO_API } = useRuntimeConfig().public
+
+const userStore = useUserStore()
+const stripeStore = useStripeStore()
+const bookStoreApiStore = useBookStoreApiStore()
+const walletStore = useWalletStore()
+
+const toast = useToast()
+const router = useRouter()
+
+const { wallet } = storeToRefs(walletStore)
+const { userLikerInfo } = storeToRefs(userStore)
+
+const error = ref('')
+const isStripeConnectReady = ref(false)
+const bookList = ref([])
+
+const channelId = computed(() => {
+  if (userLikerInfo.value?.user) {
+    return convertLikerIdToChannelId(userLikerInfo.value?.user)
+  }
+  return ''
+})
+
+const isAffiliationReady = computed(() => isStripeConnectReady.value && channelId.value)
+
+const tableColumns = [
+  {
+    key: 'image',
+    label: 'Cover',
+    sortable: false
+  },
+  {
+    key: 'className',
+    label: 'Class Name',
+    sortable: true
+  },
+  {
+    key: 'author',
+    label: 'Author',
+    sortable: true
+  },
+  {
+    key: 'priceInUSD',
+    label: 'Price in USD',
+    sortable: true
+  }, {
+    key: 'url',
+    label: isAffiliationReady.value ? 'Affiliation Link' : 'Link',
+    sortable: false
+  }
+]
```

Comment:
I think this will not dynamically update if the computed property changes, as objects are not reactive.
```suggestion
const tableColumns =computed(() => [
  {
    key: 'image',
    label: 'Cover',
    sortable: false
  },
  {
    key: 'className',
    label: 'Class Name',
    sortable: true
  },
  {
    key: 'author',
    label: 'Author',
    sortable: true
  },
  {
    key: 'priceInUSD',
    label: 'Price in USD',
    sortable: true
  }, {
    key: 'url',
    label: isAffiliationReady.value ? 'Affiliation Link' : 'Link',
    sortable: false
  }
])
```